### PR TITLE
[cherry-pick] NuGetAuthenticate: Fix bug where wrong error message when specifying …

### PR DESCRIPTION
**Pull request by zarenner**
…an ApiKey based service connection

### CODE REVIEW

File: Tasks/NuGetAuthenticateV0/task.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 0,
         "Minor": 156,
-        "Patch": 1
+        "Patch": 3
```
Comment (zarenner) [inline Tasks/NuGetAuthenticateV0/task.json]: @aasim Looks like your maven PR (which bumped the versions) wasn't cherry-picked to 156 yet?

**Comment by zarenner** [comment]
@shubham90 thanks for the review. Thoughts on if I should wait for @aasim's changes to go in or not before merging this? This branching strategy seems crazy with the multiple version number bumps from different common changes.

**Comment by shubham90** [comment]
@zarenner I think you can merge your changes and Aasim can bump it again and merge his. Yeah this branching strategy is not good but it used to be much worse than this.

**Comment by zarenner** [comment]
@shubham90 this PR already includes 2 bumps, one for his one for mine. So either I should only bump by one instead of two, or he shouldn't bump. Definitely want to make sure the version in m156 doesn't exceed master.

**Comment by shubham90** [comment]
> @shubham90 this PR already includes 2 bumps, one for his one for mine. So either I should only bump by one instead of two, or he shouldn't bump. Definitely want to make sure the version in m156 doesn't exceed master.

Yeah that makes sense. Also, you can always make master to be 0.157.0. So, the release branch will always be less than that.

**Comment by zarenner** [comment]
@aasim I am including only "my" version bump. Please bump versions when you cherry-pick your changes.

File: Tasks/NuGetAuthenticateV0/task.json (hunk 1 above)
Comment (aasim) [reply Tasks/NuGetAuthenticateV0/task.json]: Its in review right now, I have to port your changes to the merge as well, and probably bump these task versions too.

**Pull request merged**